src/_pytest/pytester.py

```diff
@@ -687,11 +698,41 @@ def getinicfg(self, source):
         return py.iniconfig.IniConfig(p)["pytest"]
 
     def makepyfile(self, *args, **kwargs):
-        """Shortcut for .makefile() with a .py extension."""
+        r"""Shortcut for .makefile() with a .py extension
+        defaults to the test name .py e.g test_foobar.py
+        Calling without kwargs will result in overwriting the initial file contents
+
+        Examples:
+
+        .. code-block:: python
+
+            def test_something(testdir):
+                # initial file is created test_something.py
+                testdir.makepyfile("foobar")
+                # to create multiple files, pass kwargs accordingly
+                testdir.makepyfile(custom="foobar")
+                # outcome is test_something.py & custom.py
+
+        """
         return self._makefile(".py", args, kwargs)
 
     def maketxtfile(self, *args, **kwargs):
-        """Shortcut for .makefile() with a .txt extension."""
+        r"""Shortcut for .makefile() with a .txt extension
+        defaults to the test name .txt e.g test_foobar.txt
+        Calling without kwargs will result in overwriting the initial file contents
```

Comment:
I think we can remove this phrase, because it conveys the impression that by calling with kwargs initial file contents are somehow not overwritten.